src/Symfony/Component/Security/Core/Authorization/AuthorizationChecker.php

```diff
@@ -24,20 +24,28 @@
  */
 class AuthorizationChecker implements AuthorizationCheckerInterface
 {
+    private array $accessDecisionStack = [];
+
     public function __construct(
         private TokenStorageInterface $tokenStorage,
         private AccessDecisionManagerInterface $accessDecisionManager,
     ) {
     }
 
-    final public function isGranted(mixed $attribute, mixed $subject = null): bool
+    final public function isGranted(mixed $attribute, mixed $subject = null, ?AccessDecision $accessDecision = null): bool
```

Comment:
Note that we *could* decide to not expose AccessDecision objects when using (User)AuthorizationCheckerInterface.
Any opinions on this?

Reply:
Having the AccessDecision available to `isGranted[ForUser]()` methods would allow userland to get the audit log more easily. It's also used right now in `AbstractController::denyUnlessGranted()`, to compute a better error message. Using the AccessDecisionManager instead of the AuthorizationChecker would be the alternative.

Reply:
Having it in `(User)AuthorizationCheckerInterface` looks useful enough